packages/devtools/devtools/src/panels/mesh/SignalMessages.tsx

```diff
@@ -104,67 +105,49 @@ const types: ColumnType<SignalResponse>[] = [
       }
     ]
   }
-];
+] as const; // this is ok because getView below will fail typecheck if this array is misdefined
 
-const getType = (id: string): ColumnType<SignalResponse> => types.find((type) => type.id === id)!;
+export type ViewType = typeof views[number]['id'];
 
-const SignalMessages = () => {
-  const devtoolsHost = useDevtools();
-  const [signalResponses, setSignalResponses] = useState<SignalResponse[]>([]);
+const getView = (id: ViewType): View<SignalResponse> => views.find((type) => type.id === id)!;
 
-  const [type, setType] = useState<ColumnType<SignalResponse>>(getType('swarm-event'));
-  const handleSearch = (text: string) => {
-    setText(text);
-  };
-
-  const [text, setText] = useState<string>();
-  const handleSelect = (id?: string) => {
-    if (id) {
-      setType(getType(id));
-    }
-  };
-
-  const defaultSubFilter = (match?: string) => (object: SignalResponse) => {
-    if (!match) {
-      return true;
-    }
-    return JSON.stringify(object).includes(match);
-  };
-
-  const getFilteredData = () => {
-    return signalResponses.filter(type.filter).filter(type.subFilter ? type.subFilter(text) : defaultSubFilter(text));
-  };
-
-  useEffect(() => {
-    const signalOutput = devtoolsHost.subscribeToSignal();
-    const signalResponses: SignalResponse[] = [];
-    signalOutput.subscribe((response: SignalResponse) => {
-      signalResponses.push(response);
-      setSignalResponses([...signalResponses]);
-    });
-
-    return () => {
-      signalOutput.close();
-    };
-  }, []);
+export type SignalMessagesProps = {
+  messages?: SignalResponse[];
+};
 
+export const SignalMessages = (props: SignalMessagesProps) => {
```

Comment:
@mykola-vrmchk check out this "pure" version of SignalMessages and it's counterpart `SignalMessagesContainer` which does 'just the data loading' for it

Reply:
this makes it easy to build inside a storybook story as opposed to in-app